optuna/samplers/_gp/sampler.py

```diff
@@ -184,18 +197,68 @@ def sample_relative(
             initial_kernel_params=self._kernel_params_cache,
             deterministic_objective=self._deterministic,
         )
+        constraints_kernel_params = [
+            gp.fit_kernel_params(
+                X=normalized_params,
+                Y=vals,
+                is_categorical=(
+                    internal_search_space.scale_types == gp_search_space.ScaleType.CATEGORICAL
+                ),
+                log_prior=self._log_prior,
+                minimum_noise=self._minimum_noise,
+                initial_kernel_params=cache,
+                deterministic_objective=self._deterministic,
+            )
+            for vals, cache in zip(
+                standardized_constraint_vals.T,
+                (
+                    cast(list[gp.KernelParamsTensor | None], self._constraints_kernel_params_cache)
+                    if self._constraints_kernel_params_cache is not None
+                    else [None] * constraint_vals.shape[1]
+                ),
+            )
+        ]
+
         self._kernel_params_cache = kernel_params
+        self._constraints_kernel_params_cache = constraints_kernel_params
 
         acqf_params = acqf.create_acqf_params(
             acqf_type=acqf.AcquisitionFunctionType.LOG_EI,
             kernel_params=kernel_params,
             search_space=internal_search_space,
             X=normalized_params,
-            Y=standarized_score_vals,
+            Y=standardized_score_vals,
+            max_Y=np.max(standardized_score_vals[np.all(constraint_vals <= 0, axis=1)]),
```

Comment:
Can we define something like this?

```python
is_feasible = np.all(constraint_vals <= 0, axis=1)
is_all_infeasible = not np.any(is_feasible)
```

Btw, `max_Y` should be `-inf` when `is_all_infeasible=True`.
Or you could alternatively ignore `f_val` in `eval_acqf` if `is_all_infeasible=True`.